core/lib/cart/index.ts

```diff
@@ -11,16 +11,38 @@ import { TAGS } from '~/client/tags';
 import { MissingCartError } from './error';
 
 export async function getCartId(): Promise<string | undefined> {
+  const anonymousSession = await getAnonymousSession();
+
+  if (anonymousSession) {
+    return anonymousSession.user?.cartId ?? undefined;
+  }
+
   const session = await auth();
 
   return session?.user?.cartId ?? undefined;
 }
 
 export async function setCartId(cartId: string): Promise<void> {
+  const anonymousSession = await getAnonymousSession();
+
+  if (anonymousSession) {
+    await updateAnonymousSession({ cartId });
+
+    return;
+  }
+
   await updateSession({ user: { cartId } });
 }
 
 export async function clearCartId(): Promise<void> {
```

Comment:
What is the point of this anymore, if we destroy anonymous session on user login?

Reply:
Took care of this in https://github.com/bigcommerce/catalyst/pull/2368